2242_Belyakov_Roman_27092022_DZ_1

romik1981 opened the pull request:
Домашнее задание по теме алгоритмы Python к уроку 1

romik1981 committed:
Update task_1.py

1 задание выполненою.

romik1981 committed:
Update_task_2

romik1981 committed:
Update_task_2-3

romik1981 committed:
Update_task_4-5

romik1981 committed:
Update_task_6-7

romik1981 committed:
done_task_1

DmitryChitalov commented on Урок 1. Практическое задание/task_1.py:
```diff
@@ -22,10 +22,10 @@ def check_1(lst_obj):
     Создать множество из списка
 
-    Сложность: !!!.
+    Сложность: T(n) = len(n) + 1, O(len(n)) зависит от длины списка list_obj.
```
Сложность: T(n) = len(n) + 1, O(len(n)) зависит от длины списка list_obj.
какая это сложность? константная?

DmitryChitalov commented on Урок 1. Практическое задание/task_1.py:
```diff
@@ -37,12 +37,12 @@ def check_2(lst_obj):
-            return False                   # !!!
-    return True                            # !!!
+    for j in range(len(lst_obj)):          # O(n)
+        if lst_obj[j] in lst_obj[j+1:]:    # O(1) + O(n) + O(n - (j+1))
```
if lst_obj[j] in lst_obj[j+1:]:    # O(1) + O(n) + O(n - (j+1))
неверно

DmitryChitalov commented on Урок 1. Практическое задание/task_1.py:
```diff
@@ -37,12 +37,12 @@ def check_2(lst_obj):
     в оставшихся справа элементах
 
-    Сложность: !!!.
+    Сложность: T(n) = n + 1 + n + n + 1 + 1 = 3n + 1 = n, O(n).
```
итоговая другая

DmitryChitalov commented on Урок 1. Практическое задание/task_1.py:
```diff
@@ -53,14 +53,14 @@ def check_3(lst_obj):
-        if lst_copy[i] == lst_copy[i+1]:     # !!!
-            return False                     # !!!
-    return True                              # !!!
+    lst_copy = list(lst_obj)                 # O(len(lst_obj)) зависит от длины списка lst_obj
```
lst_copy = list(lst_obj)                 # O(len(lst_obj)) зависит от длины списка lst_obj
какая это сложность?

DmitryChitalov commented on Урок 1. Практическое задание/task_2.py:
```diff
@@ -14,4 +14,43 @@
+num = [num for num in range(0, 100)]
+num_list = list(random.sample(num, n_num))
+print(num_list)
+
```
"квадротичный" - квадратичный

DmitryChitalov commented on Урок 1. Практическое задание/task_2.py:
```diff
@@ -14,4 +14,43 @@
+print(min_of_list_2(user_list))
+# O(1) + O(n) * (O(1) + O(1)) + O(1) = O(n)
+
+# Результат выполнения функций одинаков
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_3.py:
```diff
@@ -17,3 +17,43 @@
+
+# 1 способ
+list_values = sorted(list(companies.values()), reverse=True)[:3]                     # O(n)
+for el in list_values:                                                               # O(1)
```
for el in list_values:                                                               # O(1)
здесь линейная

DmitryChitalov commented on Урок 1. Практическое задание/task_3.py:
```diff
@@ -17,3 +17,43 @@
+companies = {'ARS': 526879.45, 'SELECT': 895674.32, 'FOX': 789456.78, 'mail-group': 478956.56, 'MSI': 478955.55}
+
+# 1 способ
+list_values = sorted(list(companies.values()), reverse=True)[:3]                     # O(n)
```
list_values = sorted(list(companies.values()), reverse=True)[:3]                     # O(n)
здесь sorted какая?

DmitryChitalov commented on Урок 1. Практическое задание/task_4.py:
```diff
@@ -22,3 +22,51 @@
+
+# Второй способ более сложный по исполнению, но позволяет ролучить больше данных которые можно использовать в далинейшей
+# разработке проекта и позволяет определить незарегестрированных пользователей.
+
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_5.py:
```diff
@@ -17,3 +17,56 @@
+
+    print(stack_1.pop_out())
+
+    print(stack_1.elems)
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_6.py:
```diff
@@ -17,3 +17,69 @@
+
+    print(solv_task.size()) # -> 2
+
+    print(rev_tasks.size())  # -> 1
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_7.py:
```diff
@@ -13,3 +13,27 @@
+
+print(pal_checker("топот"))
+print(pal_checker("молоко делили ледоколом"))
+print(pal_checker("А роза упала на лапу Азора"))
```
выполнено

romik1981 committed:
Update task_3.py

Задание №3 исправлено!
Если сложность среза n + k , то какая будет сложность если зависит от длины списка?

romik1981 committed:
Update task_2.py

Исправлено "квадротичный" - квадратичный.

romik1981 committed:
Update task_1.py

 O(len(lst_obj)) зависит от длины списка list_obj. Не знаю какая это сложность??? Думаю константная O(n)!

romik1981 committed:
Merge remote-tracking branch 'origin/lesson_1' into lesson_1

romik1981 committed:
update_task_1_2_3
